src/config/readConfigFile.ts

```diff
@@ -5,13 +5,36 @@
 import fs from "fs";
 import { buildReadConfigFileCallback } from "./buildReadConfigFileCallback";
 import { OpenResult } from "./openConfig";
+import path from "path";
+import { CheckedConfig } from "./execute";
 
 // eslint-disable-next-line functional/no-return-void, @typescript-eslint/prefer-readonly-parameter-types
 export type Accept = (value: OpenResult | PromiseLike<OpenResult>) => void;
 
 export const configFile = ".folder-structure-lint.json";
-export const message = `Could not open the config file "${configFile}"`;
+export const jsConfigFile = ".folder-structure-lint.js";
+export const message = `Could not open the config file "${configFile}" or "${jsConfigFile}"`;
 
+type MaybeCheckedConfig = Partial<CheckedConfig>;
+
+// eslint-disable-next-line max-lines-per-function
 export function readConfigFile(accept: Accept) {
-  fs.readFile(configFile, "utf8", buildReadConfigFileCallback(accept));
+  if (fs.existsSync(jsConfigFile)) {
+    const absolutePath = path.resolve(jsConfigFile);
+
+    // eslint-disable-next-line functional/no-try-statement
```

Comment:
Even once I do clean up the code, there will be no way around a `try/catch` statement around the `require` function. So, I have no choice but to ignore the eslint comment unless you can think of a better way

Reply:
I think that the try catch is good. An async import and .then just to avoid it would be unnecessary. And I can’t think of any benefit to avoiding a try catch here.